ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/feedbackController/taskspace/OrientationFeedbackController.java

```diff
@@ -631,6 +631,12 @@ public void computeIntegralTerm(FrameVector3D feedbackTermToPack)
          yoErrorRotationVectorIntegrated.setToZero(trajectoryFrame);
       }
 
+      if (yoErrorOrientation.getReferenceFrame() != trajectoryFrame)
+      {
+         yoErrorOrientation.setToZero(trajectoryFrame);
+         yoErrorRotationVectorIntegrated.setToZero(trajectoryFrame);
+      }
```

Comment:
I'm confused by this. How would it ever reach this point? If the error was in a different frame, how did the error get calculated in that frame in the PD controller portion?

Reply:
I'm not sure though, this if statement is only triggered through the orientationTest: comapreAgainstSpatialController. 
Fist of all, the structure of the orientationfeedbackController is differ with orientation part of the spatialController. 
With the robot simulation, I've never seen this if statement is triggered. 
For now, I just pushed. But later, I'll look into it again when I have spare time.